File: Java/Easy/Array/TwoSum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> counterMap = new HashMap<Integer,Integer>();
        int[] result = new int[2];
        for(int i=0;i<nums.length;i++)
        {
        	if(counterMap.containsKey(nums[i]))
        	{
        		result[0] = counterMap.get(nums[i]);
        		result[1] = i;
        		return result;
        	}
        	counterMap.put(target-nums[i],i);
        }
        return result;
    }
}

File: CSharp/Easy/Array/TwoSum.cs
/*
Example:

Given nums = [2, 7, 11, 15], target = 9,

Because nums[0] + nums[1] = 2 + 7 = 9,
return [0, 1].
*/
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int,int> diffMap = new Dictionary<int,int>();
        for(int i=0;i<nums.Length;i++)
        {
        	if(diffMap.ContainsKey(nums[i]))
        	{
        		return new int[]{diffMap[nums[i]],i};
        	}
        	else
        	{
        		if(!diffMap.ContainsKey(target-nums[i]))
        		{
        			diffMap.Add(target-nums[i],i);
        		}
        	}
        }
        return new int[]{};
    }
}